### PR TITLE
Reorganize and update PR template

### PR DESCRIPTION
**Pull request by chrisvfritz**
_Using the updated PR template below - so meta!_

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe: contributor documentation update

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the dev branch for v2.x (or to a previous version branch), _not_ the master branch
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup

**Other information:**

- Improved consistency of language in the template
- Reorganized template a little
- Added instructions at the top for how to check boxes

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,38 +3,39 @@ Please make sure to read the Pull Request Guidelines:
...
+
+- [ ] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
+- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
+- [ ] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
+- [ ] New/updated tests are included
+
+If adding a **new feature**, the PR's description includes:
+- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)
+
+If fixing a **bug**, the PR's description includes:
```
Comment (posva) [inline .github/PULL_REQUEST_TEMPLATE.md]: I find this quite cumbersome if the issue is already opened with all these details. Isn't the `When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number` enough?

Comment (chrisvfritz) [inline .github/PULL_REQUEST_TEMPLATE.md]: What if we changed these two sub-headings to, "the PR's description or a linked to issue includes"?

Comment (posva) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think that we should remove the whole block. I'm worried about the list being too long and scary for new contributors. Do we have PRs that try to fix bugs that doesn't exist?

Comment (chrisvfritz) [inline .github/PULL_REQUEST_TEMPLATE.md]: You look at PRs on this repo more than I do, so if it's not solving a problem you commonly see, I agree it does make sense to remove it.

Comment (chrisvfritz) [inline .github/PULL_REQUEST_TEMPLATE.md]: Done!

**Pull request merged**